src/log-dnslog.c

```diff
@@ -214,24 +127,86 @@ static int LogDnsLogger(ThreadVars *tv, void *data, const Packet *p, Flow *f,
         dp = p->sp;
     }
 
-    DNSQueryEntry *query = NULL;
-    TAILQ_FOREACH(query, &dns_tx->query_list, next) {
-        LogQuery(aft, timebuf, dstip, srcip, dp, sp, dns_tx, query);
-    }
+    char *buf;
+    LogDnsFileCtx *hlog = aft->dnslog_ctx;
 
-    if (dns_tx->rcode)
-        LogAnswer(aft, timebuf, srcip, dstip, sp, dp, dns_tx, NULL);
-    if (dns_tx->recursion_desired)
-        LogAnswer(aft, timebuf, srcip, dstip, sp, dp, dns_tx, NULL);
+    for (uint16_t i = 0; i < 0xffff; i++) {
```

Comment:
have you done any profiling of this code? It seems a bit cleaner than the json object sharing, but it would be interesting to get some performance metrics.

Wonder if we could pass around some normalized data in larger blobs too. Not sure yet how that could/should look like. Perhaps we can look at how for example the kernel/userspace xfer for conntrack is done.